talks/introduction/intro.md

```diff
@@ -0,0 +1,59 @@
+---
+author: Ulises (Tirado Zatarain | Mendez Martinez)
+title: Introduction to Algorithms
+date: Apr, 2021
+---
+
+# Why?
+
+* Limited Resources.
+* Efficiency.
+* Just for Fun!
+
+But what is an **algorithm**?
+
+---
+
+# When?
+
+* The **first rule** of optimization is: Don't do it. The **second rule** of optimization is: Don't do it yet.
+
+---
+
+# How to compare them?
+
+* Correct vs Incorrect.
+* Well, it depends, suppose we have two algorithms:
+  * One works better for large inputs.
+  * One works better for small inputs.
+  * Which one is better?
+* Another sample: Sorting Algorithms:
+  * Number of items to be sorted.
+  * Are they already somewhat sorted?
+  * item values/limits?
+* Complexity:
+  * Runtime (Temporal)
+  * Spatial (Memory)
+
+---
+
+# Interviews.
+
+* why?
+* are they even used in daily work?
+
+---
+
+# But technology has improved!
+
+* Moore's law can't continue forever.
+* With better technology, bigger challenges.
+* Parallelism.
+
```

Comment:
I would add here another bullet point with Machine Learning and IA, is another common "excuse" I heard about, so for some reason there are people whose think either IA doesn't have complexity or doesn't have underlying classic algorithms or whatever.